### PR TITLE
Replace jax tree manipulation method that are being deprecated with jax.tree_util equivalents

### PR DESCRIPTION
**Pull request by mahdi-shafiei**

### CODE REVIEW

**Comment by patrick-kidger** [comment]
Looks like the pre-commit checks failed. See CONTRIBUTING.md.

I'd suggest doing `import jax.tree_util as jtu` for brevity.

**Comment by mahdi-shafiei** [comment]
> Looks like the pre-commit checks failed. See CONTRIBUTING.md.
> 
> I'd suggest doing `import jax.tree_util as jtu` for brevity.

Looks like I was able to push my changes. Pre-commit checks pass now.

File: diffrax/misc/omega.py [hunk 1]
```diff
@@ -36,7 +36,8 @@ def __init__(self, value, is_leaf=None):
         **Arguments:**
 
         - `value`: The PyTree to wrap.
-        - `is_leaf`: An optional value for the `is_leaf` argument to `jax.tree_map`.
+        - `is_leaf`: An optional value for the `is_leaf` argument to
+          `jtu.tree_map`.
```
Comment (patrick-kidger) [inline diffrax/misc/omega.py:40]: In docs can this still be spelled out as `jax.tree_utils.tree_map`.

File: docs/devdocs/omega.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ Other operations are of course defined: `ω` understands several of the built-in
 
 ### Non-goals
 
-Making anything like `jax.numpy.maximum(x**ω, y**ω)` work is not a goal for `ω`. Just use a regular `jax.tree_map` in these situtions. `ω` only aims to support overloadable Python operations, and as a convenience single-argument functions via e.g. `ω(x).call(jax.numpy.abs)`.
+Making anything like `jax.numpy.maximum(x**ω, y**ω)` work is not a goal for `ω`. Just use a regular `jtu.tree_map` in these situtions. `ω` only aims to support overloadable Python operations, and as a convenience single-argument functions via e.g. `ω(x).call(jax.numpy.abs)`.
```
Comment (patrick-kidger) [inline docs/devdocs/omega.md:35]: doc here too

File: examples/continuous_normalising_flow.ipynb [hunk 3]
```diff
@@ -748,7 +749,12 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.9.7"
+   "version": "3.10.4"
+  },
+  "vscode": {
+   "interpreter": {
+    "hash": "f2e7013ef9ffca27b94c39f5eba58524f9e175f876d312983be1369a16a681ac"
+   }
```
Comment (patrick-kidger) [inline examples/continuous_normalising_flow.ipynb:757]: I admire your keenness to rerun all the notebooks to be sure they still run, but I'd rather have a smaller diff to review -- it's much harder to review changes to images in `.ipynb` files.

IMO for little changes it's totally fine to just edit the JSON structure of the code directly. (The pre-commit hooks should verify at least basic checks like syntactic well-formedness of the code produced.)

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+    - id: nbqa-isort 
+    - id: nbqa-flake8
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.10.1
+    hooks:
+    - id: isort
+  - repo: https://github.com/pycqa/flake8
+    rev: 4.0.1
+    hooks:
+    - id: flake8
```
Comment (patrick-kidger) [inline .pre-commit-config.yaml:19]: What change is actually being made to this file? If this is just some git oddity and this file was meant to be left untouched, then can it be removed from this commit.

File: docs/devdocs/omega.md (hunk 2 above)
Comment (mahdi-shafiei) [reply docs/devdocs/omega.md]: done

File: examples/continuous_normalising_flow.ipynb (hunk 3 above)
Comment (mahdi-shafiei) [reply examples/continuous_normalising_flow.ipynb]: Makes sense, done.

File: .pre-commit-config.yaml (hunk 4 above)
Comment (mahdi-shafiei) [reply .pre-commit-config.yaml]: I don't know why git picked it up. I removed it from the commit.

**Pull request merged**